### PR TITLE
sample: syst: Add config for testing on intel_adsp_cavs platforms

### LINKED ISSUE #43344: intel_adsp_cavs25: samples/subsys/logging/syst is failing with a timeout when the sample is enabled to run on intel_adsp_cavs25
**Describe the bug**
samples/subsys/logging/syst is failing with a timeout as the switching to sys-t raw data doesn't happen and logger is waiting. 

Please also mention any information which could help others to understand
the problem you're facing:
- happens on any of the cavs hardware once they are included in the platform allow list. 

**To Reproduce**
1.  add the intel_adsp_cavs15 or cavs18 or cavs25 to sample.yaml
```
sample:
  name: syst sample
tests:
  sample.logger.syst.v1.deferred:
    extra_args: OVERLAY_CONFIG=overlay_deferred.conf
    platform_allow: mps2_an385 qemu_x86 sam_e70_xplained intel_adsp_cavs15
    tags: logging
    harness: console
    harness_config:
      type: one_line
      regex:
        - "SYS-T RAW DATA: "

```
2. twister -W -p intel_adsp_cavs15 --device-testing --device-serial-pty=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh --west-flash=$HOME/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstwist.sh -T samples/subsys/logging/syst

3. see below error. 

**Expected behavior**
Test should execute successfully similar to qemu_x86.

**Logs and console output**
```
DEBUG   - DEVICE: INFO:cavs-fw:Mapped PCI bar 0 of length 16384 bytes.
DEBUG   - DEVICE: INFO:cavs-fw:Selected output stream 7 (GCAP = 0x1006701)
DEBUG   - DEVICE: INFO:cavs-fw:Mapped PCI bar 4 of length 1048576 bytes.
DEBUG   - DEVICE: INFO:cavs-fw:Detected cAVS 1.5 hardware
DEBUG   - DEVICE: INFO:cavs-fw:Trimming 16 bytes of extended manifest
DEBUG   - DEVICE: INFO:cavs-fw:Resetting HDA device
DEBUG   - DEVICE: INFO:cavs-fw:Powering down DSP cores
DEBUG   - DEVICE: INFO:cavs-fw:Configuring HDA stream 7 to transfer firmware image
DEBUG   - DEVICE: INFO:cavs-fw:Mapped 2M huge page at 0x150600000 to contain 110592 bytes of firmware
DEBUG   - DEVICE: INFO:cavs-fw:Filled the buffer descriptor list (BDL) for DMA.
DEBUG   - DEVICE: INFO:cavs-fw:Starting DSP, ADSPCS = 0x303
DEBUG   - DEVICE: INFO:cavs-fw:Wait for ROM startup
DEBUG   - DEVICE: INFO:cavs-fw:Sending IPC command, HIPCR = 0x81004e00
DEBUG   - DEVICE: INFO:cavs-fw:Starting DMA, FW_STATUS = 0x5000001
DEBUG   - DEVICE: INFO:cavs-fw:Waiting for firmware handoff, FW_STATUS = 0x5000001
DEBUG   - DEVICE: INFO:cavs-fw:cAVS firmware load complete
DEBUG   - DEVICE: --
DEBUG   - DEVICE: [00000062] <err> syst: Error message example.
DEBUG   - DEVICE: [00000062] <wrn> syst: Warning message example.
DEBUG   - DEVICE: [00000062] <inf> syst: Info message example.
DEBUG   - DEVICE: [00000062] <dbg> syst: Debug message example.
DEBUG   - DEVICE: [00000062] <dbg> syst: Debug message example, 1
DEBUG   - DEVICE: [00000063] <dbg> syst: Debug message example, 1, 2
DEBUG   - DEVICE: [00000063] <dbg> syst: Debug message example, 1, 2, 3
DEBUG   - DEVICE: [00000063] <dbg> syst: Debug message example, 1, 2, 3, 0x4
DEBUG   - DEVICE: [00000063] <dbg> syst: char !
DEBUG   - DEVICE: [00000063] <dbg> syst: s str static str
DEBUG   - DEVICE: [00000063] <dbg> syst: d str dynamic str
DEBUG   - DEVICE: [00000063] <dbg> syst: mixed str dynamic str --- dynamic str --- another dynamic str --- another dynamic str
DEBUG   - DEVICE: [00000063] <dbg> syst: mixed c/s ! static str dynamic str static str !
DEBUG   - DEVICE: [00000063] <dbg> syst: Debug message example, %f
DEBUG   - DEVICE: [00000063] <err> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000064] <wrn> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000064] <inf> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000064] <dbg> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: hello sys-t on board intel_adsp_cavs15
DEBUG   - DEVICE: [00000064] <err> syst: Error message example.
DEBUG   - DEVICE: [00000064] <wrn> syst: Warning message example.
DEBUG   - DEVICE: [00000064] <inf> syst: Info message example.
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example.
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example, 1
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example, 1, 2
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example, 1, 2, 3
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example, 1, 2, 3, 0x4
DEBUG   - DEVICE: [00000065] <dbg> syst: char !
DEBUG   - DEVICE: [00000065] <dbg> syst: s str static str
DEBUG   - DEVICE: [00000065] <dbg> syst: d str dynamic str
DEBUG   - DEVICE: [00000065] <dbg> syst: mixed str dynamic str --- dynamic str --- another dynamic str --- another dynamic str
DEBUG   - DEVICE: [00000065] <dbg> syst: mixed c/s ! static str dynamic str static str !
DEBUG   - DEVICE: [00000065] <dbg> syst: Debug message example, %f
DEBUG   - DEVICE: [00000065] <err> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000066] <wrn> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000066] <inf> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000066] <dbg> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: hello sys-t on board intel_adsp_cavs15
DEBUG   - DEVICE: [00000066] <err> syst: Error message example.
DEBUG   - DEVICE: [00000066] <wrn> syst: Warning message example.
DEBUG   - DEVICE: [00000067] <inf> syst: Info message example.
DEBUG   - DEVICE: [00000067] <dbg> syst: Debug message example.
DEBUG   - DEVICE: [00000067] <dbg> syst: Debug message example, 1
DEBUG   - DEVICE: [00000067] <dbg> syst: Debug message example, 1, 2
DEBUG   - DEVICE: [00000067] <dbg> syst: Debug message example, 1, 2, 3
DEBUG   - DEVICE: [00000067] <dbg> syst: Debug message example, 1, 2, 3, 0x4
DEBUG   - DEVICE: [00000067] <dbg> syst: char !
DEBUG   - DEVICE: [00000067] <dbg> syst: s str static str
DEBUG   - DEVICE: [00000067] <dbg> syst: d str dynamic str
DEBUG   - DEVICE: [00000067] <dbg> syst: mixed str dynamic str --- dynamic str --- another dynamic str --- another dynamic str
DEBUG   - DEVICE: [00000067] <dbg> syst: mixed c/s ! static str dynamic str static str !
DEBUG   - DEVICE: [00000068] <dbg> syst: Debug message example, %f
DEBUG   - DEVICE: [00000068] <err> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000068] <wrn> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000068] <inf> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000068] <dbg> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: hello sys-t on board intel_adsp_cavs15
DEBUG   - Timed out while monitoring serial output on intel_adsp_cavs15
```

**Environment (please complete the following information):**
 - OS: Fedora
 - Toolchain (e.g Zephyr SDK, ...) - SDK 13.2
 - Commit SHA: 04ea1ee1afab

### PR DESCRIPTION
**Pull request by aasthagr**
Adding configurations for enabling this test on cavs platforms.
This patch also contains a fix to mipi_syst library which resolves
the memory alignment issue across different architectures.

Fixes #43344

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

----------------
This is the old change set.

See PR #44491 for the most recent change set.

### CODE REVIEW

**Review by teburd — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by yerabolu — COMMENTED** [review]
Sorry, forgot to run the west update to pull in the mipi-sys-t changes. After doing the west update, sample.logger.syst.adsp_cavs sample passed.

File: samples/subsys/logging/syst/sample.yaml [hunk 1]
```diff
@@ -10,16 +10,19 @@ tests:
...
+        - "SYST Sample Execution Completed"
     extra_configs:
       - CONFIG_LOG1=y
   sample.logger.syst.v1.immediate:
+    extra_args: OVERLAY_CONFIG=overlay_immediate.conf
     platform_allow: mps2_an385 qemu_x86 sam_e70_xplained
     tags: logging
     harness: console
     harness_config:
       type: one_line
```
Comment (yerabolu) [inline samples/subsys/logging/syst/sample.yaml:22]: should this be a multi_line harness_config type?

File: west.yml [hunk 2]
```diff
@@ -179,7 +179,7 @@ manifest:
       path: modules/debug/mipi-sys-t
       groups:
         - debug
-      revision: d9da086b11cda494d85f4d8a9829f505c2d5e380
+      revision: pull/9/head
```
Comment (nordic-krch) [inline west.yml:182]: pr got merged

File: samples/subsys/logging/syst/sample.yaml [hunk 3]
```diff
@@ -28,14 +31,26 @@ tests:
...
     harness: console
     harness_config:
-      type: one_line
+      type: multi_line
       regex:
         - "SYS-T RAW DATA: "
+        - "SYST Sample Execution Completed"
+  sample.logger.syst.adsp_cavs:
+    extra_args: OVERLAY_CONFIG=overlay_cavs.conf
+    platform_allow: intel_adsp_cavs15 intel_adsp_cavs18 intel_adsp_cavs20 intel_adsp_cavs25
```
Comment (nashif) [inline samples/subsys/logging/syst/sample.yaml:50]: doing this you are missing all the other tests on the listed platforms. You should add the overlay in boards/<board name> instead and remove all platform_allow entries here and instead using integration_platforms to allow more coverage.

**Review by nashif — CHANGES_REQUESTED** [review]

**Pull request closed without merging**